main.py

```
@@ -604,0 +1,1133 @@
+            max_age: Maximum age of cached responses in seconds (default: 1 hour)
+        """
+        self.cache_dir = Path(cache_dir)
+        self.cache_dir.mkdir(parents=True, exist_ok=True)
+        self.max_age = max_age
+        self.memory_cache: Dict[
+            str, Tuple[str, float]
+        ] = {}  # url -> (content, timestamp)
+
+    def _get_cache_key(self, url: str) -> str:
+        """Generate a cache key from a URL."""
+        return hashlib.md5(url.encode()).hexdigest()
```

Ruff (0.8.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)